zeebe/dynamic-node-id-provider/src/main/java/io/camunda/zeebe/dynamic/nodeid/Lease.java

```diff
@@ -9,11 +9,43 @@
 
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import io.camunda.zeebe.dynamic.nodeid.repository.Metadata;
 import java.io.IOException;
 import java.time.Duration;
 import java.time.Instant;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+public record Lease(
+    String taskId, long timestamp, NodeInstance nodeInstance, VersionMappings versionMappings) {
```

Comment:
:wrench: May be rename `versionMappings` to `knownVersionMappings`, or something else to indicate that this is map of nodeId -> version that is observed by this node. 

In addition, documentation of the fields would be helpful.